src/otx/algo/keypoint_detection/rtmpose.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""RTMDet model implementations."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from otx.algo.common.backbones import CSPNeXt
+from otx.algo.common.utils.assigners import DynamicSoftLabelAssigner
+from otx.algo.common.utils.samplers import PseudoSampler
+from otx.algo.keypoint_detection.heads.rtmcc_head import RTMCCHead
+from otx.algo.keypoint_detection.losses.kl_discret_loss import KLDiscretLoss
+from otx.algo.keypoint_detection.topdown import TopdownPoseEstimator
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.model.keypoint_detection import OTXKeypointDetectionModel
+
+if TYPE_CHECKING:
+    from otx.core.exporter.base import OTXModelExporter
+    from otx.core.types.export import TaskLevelExportParameters
+
+
+class RTMPose(OTXKeypointDetectionModel):
+    """OTX Detection model class for RTMDet."""
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.image_size is None:
+            raise ValueError(self.image_size)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=self.image_size,
+            mean=self.mean,
+            std=self.std,
+            resize_mode="fit_to_window_letterbox",
+            pad_value=114,
+            swap_rgb=True,
+            via_onnx=True,
+            onnx_export_configuration={
+                "input_names": ["image"],
+                "output_names": ["points"],
+                "dynamic_axes": {
+                    "image": {0: "batch", 2: "height", 3: "width"},
+                    "points": {0: "batch", 1: "num_dets"},
+                },
+                "autograd_inlining": False,
+            },
+            output_names=["points", "feature_vector", "saliency_map"] if self.explain_mode else None,
+        )
+
+    @property
+    def _export_parameters(self) -> TaskLevelExportParameters:
+        """Defines parameters required to export a particular model implementation."""
+        return super()._export_parameters.wrap(optimization_config={"preset": "mixed"})
+
+
+class RTMPoseTiny(RTMPose):
+    """RTMPose Tiny Model."""
+
+    load_from = "https://download.openmmlab.com/mmpose/v1/projects/rtmposev1/cspnext-tiny_udp-aic-coco_210e-256x192-cbed682d_20230130.pth"
+    image_size = (1, 3, 192, 256)
+    mean = (123.675, 116.28, 103.53)
+    std = (58.395, 57.12, 57.375)
+
+    def _build_model(self, num_classes: int) -> RTMPose:
+        train_cfg = {
```

Comment:
In OTX2.2 we want to perform refactoring and actually change the current mmdetection like approach of hiding all parameters in implicit "*_cfg". Instead we will use explicit approach with passing all required parameters to base model avoiding configs. In order not to do the job twice, could you join this new approach?

Also, we would like to separate loss function form models head and use loss directly in the base model. 

I know that it can require time to refactor  this, but we are currently working on refactoring task and it will be really helpful if newly added task will be already in the right shape.

Reply:
At this moment, I have removed *_cfg (train/test) for RTMPose. I will revisit to here for moving loss to the base model level later.